test.ps1

```diff
@@ -22,9 +22,79 @@ param (
     [string]$Exercise
 )
 
-$args = if ($Exercise) { @("--exercise=$Exercise") } else { @() }
+$ErrorActionPreference = 'Stop'
 
-dotnet tool install -g Cake.Tool --version 0.33.0
-dotnet cake test.cake -- $args
+$buildDir = Join-Path $PSScriptRoot "build"
+$exercisesDir = Resolve-Path "exercises"
+
+# By default, PowerShell does not check the return code
+# of native commands (see https://github.com/PowerShell/PowerShell-RFC/pull/88/files)
+function RunCommand ($command) {
+    Invoke-Expression $command
+
+    if ($Lastexitcode -ne 0) {
+        exit $Lastexitcode 
+    }
+}
+
+function ConfigletLint {
+    Write-Output "Linting config.json"
+    RunCommand "./bin/fetch-configlet"
+    RunCommand "./bin/configlet lint ."
+}
+
+function BuildGenerators { 
+    Write-Output "Building generators"
+    RunCommand "dotnet build ./generators"
+}
+function Clean {
```

Comment:
Nit: this function and the function below don't have an empty newline before them